gravitee-am-identityprovider/gravitee-am-identityprovider-jdbc/src/main/java/io/gravitee/am/identityprovider/jdbc/JdbcAbstractProvider.java

```diff
@@ -106,4 +113,34 @@ protected void computeMetadata(Map<String, Object> claims) {
         } catch (Exception e) {
         }
     }
+
+    protected Flowable<Result> query(String sql, Object... args) {
+        return Single.fromPublisher(connectionPool.create())
+                .toFlowable()
+                .flatMap(connection ->
+                        query(connection, sql, args)
+                                .doFinally(() -> Completable.fromPublisher(connection.close()).subscribe()));
+    }
+
+    protected Flowable<Result> query(Connection connection, String sql, Object... args) {
```

Comment:
Could we make this method private to not allow using it and remove "selectUserByUsername" JdbcUserProvider the one with connection parameter in JdbcUserProvider?

Reply:
We can't make it private as it is used by the JdbcUserProvider implementation during the provider startup. 
I removed the selectUserByUsername.

Reply:
yes, that's true, but it can be removed and replaced by the query without a connection param.

Reply:
yes, but for efficiency, I do prefer keeping this code as it is.